### PR TITLE
Use yum provider for bundler on EL7 (and Fedora).

### PR DESCRIPTION
**Pull request by logic**
Simple patch that lets you a) use a yum provider for bundler, and b) defaults to it on EL7 (and as luck would have it, also catches any recent Fedora release, making this work flawlessly on Fedora 23+ for my purposes).

### CODE REVIEW

File: manifests/dev.pp [hunk 1]
```diff
@@ -49,7 +49,7 @@
   # as the package ensure covers _multiple_ packages
   # specifying a version may cause issues.
   validate_re($ensure,['^installed$', '^present$', '^absent$', '^latest$'])
-  validate_re($bundler_provider,['^gem$','^apt$'])
+  validate_re($bundler_provider,['^gem$','^apt$', '^yum$'])
```
Comment (ghoneycutt) [inline manifests/dev.pp]: README needs to be updated to include `yum`

File: manifests/params.pp [hunk 2]
```diff
@@ -26,9 +26,15 @@
         $rake_package     = 'rubygem-rake'
         $rake_provider    = 'yum'
       }
-      $bundler_ensure   = 'installed'
-      $bundler_package  = 'bundler'
-      $bundler_provider = 'gem'
+      if versioncmp($::operatingsystemmajrelease, '7') < 0 {
```
Comment (ghoneycutt) [inline manifests/params.pp]: This is confusing with the if statement above for EL5. Suggest refactoring into a case statement on `operatingsystemmajrelease` with a default case that fails for unsupported systems.

**Comment by ghoneycutt** [comment]
Hi @logic, this looks great, though is missing spec tests.

**Comment by ghoneycutt** [comment]
Suggest adding your functionality with spec tests without dropping support for Puppet v3. In another, PR with no other changes, we can move to Puppet4 and do a major version bump.

**Comment by logic** [comment]
Whoops! I updated my fork but didn't recall that I'd opened these PRs. Sorry about the noise, I'm going to close these and will re-open when I have a chance to update them properly.

**Pull request closed without merging**